pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -190,6 +185,23 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 		return nil, err
 	}
 
+	// Initialize SDN before building kubelet config so it can modify options
```

Comment:
Does NewNodePlugin start goroutines?  If it does, it cannot be in this method, or you need to split the starting goroutines part into a `Start()/Run()` method that is called later.